ui/src/CollectionExerciseList.js

```diff
@@ -0,0 +1,402 @@
+import React, { Component } from "react";
+import "@fontsource/roboto";
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  Paper,
+  TextField,
+  Typography,
+} from "@material-ui/core";
+import Table from "@material-ui/core/Table";
+import TableBody from "@material-ui/core/TableBody";
+import TableCell from "@material-ui/core/TableCell";
+import TableContainer from "@material-ui/core/TableContainer";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import {
+  getAuthorisedActivities
+} from "./Utils";
+import { Link } from "react-router-dom";
+
+class CollectionExerciseList extends Component {
+  state = {
+    authorisedActivities: [],
+    collectionExercises: [],
+    createCollectionExerciseDialogDisplayed: false,
+    newCollectionExerciseName: "",
+    newCollectionExerciseNameError: false,
+    newCollectionExerciseReference: "",
+    newCollectionExerciseReferenceError: false,
+    newCollectionExerciseStartDate: null,
+    newCollectionExerciseEndDate: null,
+    newCollectionExerciseDateError: "",
+    newCollectionExerciseMetadata: "",
+    newCollectionExerciseMetadataError: false,
+    newCollectionExerciseCIRules: "",
+    newCollectionExerciseCIRulesError: false,
+
+  };
+
+  componentDidMount() {
+    this.getAuthorisedBackendData();
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  getAuthorisedBackendData = async () => {
+    const authorisedActivities = await getAuthorisedActivities();
+    this.setState({ authorisedActivities: authorisedActivities });
+    this.refreshDataFromBackend(authorisedActivities);
+
+    this.interval = setInterval(
+      () => this.getCollectionExercises(authorisedActivities),
+      1000
+    );
+  };
+
+  refreshDataFromBackend = async (authorisedActivities) => {
+    this.getCollectionExercises(authorisedActivities);
+  };
+
+  getCollectionExercises = async (authorisedActivities) => {
+    if (!authorisedActivities.includes("LIST_COLLECTION_EXERCISES")) return;
+
+    const response = await fetch(
+      `/api/collectionExercises/?surveyId=${this.props.surveyId}`
+    );
+    const collexJson = await response.json();
+
+    this.setState({
+      collectionExercises: collexJson,
+    });
+  };
+
+  openCreateCollectionExerciseDialog = () => {
+    this.createCollectionExerciseInProgress = false;
+
+    this.setState({
+      newCollectionExerciseName: "",
+      newCollectionExerciseReference: "",
+      newCollectionExerciseStartDate: this.getTimeNowForDateTimePicker(),
+      newCollectionExerciseEndDate: this.getTimeNowForDateTimePicker(),
+      newCollectionExerciseMetadata: "",
+      newCollectionExerciseNameError: false,
+      newCollectionExerciseReferenceError: false,
+      createCollectionExerciseDialogDisplayed: true,
+      newCollectionExerciseMetadataError: false,
+      newCollectionExerciseDateError: "",
```

Comment:
Should be a boolean instead of string

Reply:
newCollectionExerciseDateError  Used as a string everywhere?

Reply:
if we just want to mark a field as having an error, we would use a boolean. If we want to display some "error helper text" then we can have it as a string. Empty string = falsy. Populated string = truthy. So we don't need an extra boolean